model/driver/src/icon4py/model/driver/initialization_utils.py

```diff
@@ -559,3 +570,33 @@ def configure_logging(
     console_handler.setFormatter(formatter)
     console_handler.setLevel(logging_level)
     logging.getLogger("").addHandler(console_handler)
+
+
+def create_grid_info(
+    grid_file: str,
+) -> tuple[icon_grid.GridShape, uuid.UUID]:
+    """
+    Create grid shape and its uuid.
+
+    Args:
+        grid_file: path of the grid file
+
+    Returns:
+        grid_shape: grid shape containing the root, level, and geometry type
+        grid_uuid: id (uuid) of the horizontal grid
+    """
+    grid = nc4.Dataset(grid_file, "r", format="NETCDF4")
+    grid_root = grid.getncattr("grid_root")
+    grid_level = grid.getncattr("grid_level")
+    grid_uuid = uuid.UUID(grid.getncattr("uuidOfHGrid"))
```

Comment:
We changed the type of the id to `str` recently.
```suggestion
    grid_uuid = grid.getncattr("uuidOfHGrid")
```